server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorDVLeafFieldData.java

```diff
@@ -76,4 +86,94 @@ public DocValuesScriptFieldFactory getScriptFieldFactory(String name) {
         }
     }
 
+    @Override
+    public FormattedDocValues getFormattedValues(DocValueFormat format) {
+        return switch (elementType) {
+            case BYTE, BIT -> new DenseVectorNumericValues(elementType == ElementType.BIT ? this.dims / Byte.SIZE : this.dims) {
+                private byte[] values;
+                private ByteVectorValues byteVectorValues; // use when indexed
+
+                @Override
+                public boolean advanceExact(int docId) throws IOException {
+                    if (indexed) {
+                        if (byteVectorValues == null) {
+                            this.byteVectorValues = reader.getByteVectorValues(field);
+                        }
+                        if (iterator == null) {
+                            this.iterator = byteVectorValues.iterator();
+                        }
+                        if (iterator.advance(docId) == NO_MORE_DOCS) {
+                            return false;
+                        }
+                        values = byteVectorValues.vectorValue(iterator.index());
+                    } else {
+                        if (binary == null) {
+                            binary = DocValues.getBinary(reader, field);
+                        }
+                        if (binary.advance(docId) == NO_MORE_DOCS) {
+                            return false;
+                        }
+                        BytesRef ref = binary.binaryValue();
+                        ByteBuffer byteBuffer = ByteBuffer.wrap(ref.bytes, ref.offset, ref.length);
+                        if (indexVersion.onOrAfter(DenseVectorFieldMapper.LITTLE_ENDIAN_FLOAT_STORED_INDEX_VERSION)) {
+                            byteBuffer.order(ByteOrder.LITTLE_ENDIAN);
+                        }
+                        values = new byte[dims];
+                        for (int dim = 0; dim < dims; dim++) {
+                            values[dim] = byteBuffer.get();
+                        }
+                    }
+                    valuesCursor = 0;
+                    return true;
+                }
+
+                public Object nextValue() {
+                    return values[valuesCursor++];
+                }
+            };
+            case FLOAT -> new DenseVectorNumericValues(dims) {
+                private float[] values;
+                private FloatVectorValues floatVectorValues; // use when indexed
+
+                @Override
+                public boolean advanceExact(int docId) throws IOException {
+                    if (indexed) {
+                        if (floatVectorValues == null) {
```

Comment:
can we have 1 `if` statement: 

```java
if (floatVectorValues == null) {
   floatVectorValues = reader.getFloatVectorValues(field);
    iterator = floatVectorValues.iterator();
}
```
iterator is coupled with floatVectorValues and you must modify it when floatVectorvalues change.

Reply:
Understood, will do.